src/app/tabbed-menu.js

```diff
@@ -0,0 +1,48 @@
+var $ = require('jquery')
+
+module.exports = tabbedMenu
+
+function tabbedMenu (compiler, loadingSpinner, self) {
+  $('#options li').click(function (ev) {
+    var $el = $(this)
+    selectTab($el)
+  })
+
+  // initialize tabbed menu
+  selectTab($('#options .envView'))
+
+  // add event listeners for loading spinner
+  // compiler.event.register('compilationStarted', function compilationStarted () {
+  compiler.event.register('loadingCompiler', function compilationStarted () {
+    var contractTab = document.querySelector('.envView')
+    if (!contractTab.children.length) {
+      contractTab.appendChild(loadingSpinner(function cb (finish) {
+        // compiler.event.register('compilationFinished', function () {
+        compiler.event.register('compilerLoaded', finish)
```

Comment:
better to register to `compilerLoaded` outside of the `loadingCompiler` registration